base/path.jl

```diff
@@ -196,6 +197,32 @@ function pathsep(paths::AbstractString...)
     return path_separator
 end
 
+"""
+    splitpath(path::AbstractString) -> Vector{String}
+
+Split a file path into all its path components. This is the opposite of
+`joinpath`. Returns an array of substrings, one for each directory or file in
+the path, including the root directory if present.
+
+# Examples
+```jldoctest
+julia> splitpath("/home/myuser/example.jl")
+["/", "home", "myuser", "example.jl"]
+```
+"""
+function splitpath(p::String)
+    out = String[]
+    while !isempty(p)
+        dir, base = splitdir(p)
+        dir == p && (push!(out, dir); break)  # Reached root node.
+        if !isempty(base)  # Skip trailing '/' in basename
+            push!(out, base)
```

Comment:
`pushfirst!` and skip the `reverse!` is clearer IMO (as suggested earlier).

Reply:
Done.